kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -380,6 +381,7 @@ object Fiber extends FiberPlatformSpecific:
                                             // Store successful result and its original index for ordering
                                             indices(okInt - 1) = idx
                                             results(okInt - 1) = v.asInstanceOf[AnyRef]
+                                            discard(VarHandle.storeStoreFence())
```

Comment:
I had trouble with flaky tests in Native. The fence was already technically missing here but the JVM probably introduces fences right after so the bug wasn't visible. That doesn't seem the case for Native.